src/main/java/uk/gov/justice/digital/config/JobArguments.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.spark.streaming.Duration;
+import org.apache.spark.streaming.Durations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Class that defines and provides access to the job arguments that we support.
+ *
+ * Arguments are taken from the values parsed by the CommandLinePropertySource.
+ */
+@Singleton
+public class JobArguments {
+
+    private static final Logger logger = LoggerFactory.getLogger(JobArguments.class);
+
+    public static final String AWS_DYNAMODB_ENDPOINT_URL = "dpr.aws.dynamodb.endpointUrl";
+    public static final String AWS_KINESIS_ENDPOINT_URL = "dpr.aws.kinesis.endpointUrl";
+    public static final String AWS_REGION = "dpr.aws.region";
+    public static final String CURATED_S3_PATH = "dpr.curated.s3.path";
+    public static final String DOMAIN_CATALOG_DATABASE_NAME = "dpr.domain.catalog.db";
+    public static final String DOMAIN_NAME = "dpr.domain.name";
+    public static final String DOMAIN_OPERATION = "dpr.domain.operation";
+    public static final String DOMAIN_REGISTRY = "dpr.domain.registry";
+    public static final String DOMAIN_TARGET_PATH = "dpr.domain.target.path";
+    public static final String DOMAIN_TABLE_NAME = "dpr.domain.table.name";
+    public static final String KINESIS_READER_BATCH_DURATION_SECONDS = "dpr.kinesis.reader.batchDurationSeconds";
+    public static final String KINESIS_READER_STREAM_NAME = "dpr.kinesis.reader.streamName";
+    public static final String RAW_S3_PATH = "dpr.raw.s3.path";
+    public static final String STRUCTURED_S3_PATH = "dpr.structured.s3.path";
+    public static final String VIOLATIONS_S3_PATH = "dpr.violations.s3.path";
+
+    private final Map<String, String> config;
+
+    @Inject
+    public JobArguments(ApplicationContext context) {
+        this(getCommandLineArgumentsFromContext(context));
+    }
+
+    public JobArguments(Map<String, String> config) {
+        this.config = config.entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        logger.info("Job initialised with parameters: {}", config);
+    }
+
+    public String getAwsRegion() {
+        return getArgument(AWS_REGION);
+    }
+
+    public String getAwsKinesisEndpointUrl() {
+        return getArgument(AWS_KINESIS_ENDPOINT_URL);
+    }
+
+    public String getAwsDynamoDBEndpointUrl() {
+        return getArgument(AWS_DYNAMODB_ENDPOINT_URL);
+    }
+
+    public String getKinesisReaderStreamName() {
+        return getArgument(KINESIS_READER_STREAM_NAME);
+    }
+
+    public Duration getKinesisReaderBatchDuration() {
+        val durationSeconds = getArgument(KINESIS_READER_BATCH_DURATION_SECONDS);
+        val parsedDuration = Long.parseLong(durationSeconds);
+        return Durations.seconds(parsedDuration);
```

Comment:
keep it simple in one single line: Durations.seconds(Long.parseLong(getArgument(KINESIS_READER_BATCH_DURATION_SECONDS)))